examples/ucp_client_server.c

```diff
@@ -462,6 +462,7 @@ ucs_status_t ucp_am_data_cb(void *arg, const void *header, size_t header_length,
 
     if ((header != NULL) || (header_length != 0)) {
         fprintf(stderr, "received unexpected header, length %ld", header_length);
+        return UCS_ERR_UNSUPPORTED;
```

Comment:
would it really fail the CI?
it should be abort() or something here.. or set some global var to make sure process exits with error

Reply:
It would fail the CI (through `ucs_assert(status == UCS_OK)`). Anyway, replaced with `abort()`.